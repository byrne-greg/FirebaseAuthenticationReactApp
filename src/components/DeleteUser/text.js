export default {
  deleteYourAccount: "Delete your account",
  deleteYourAccountQuestion: "Are you sure you want to delete your account?",
  deleteAccount: "Delete account",
  accountDeleted: "Account deleted",
  accountIsDeleted: "You have successfully deleted your account"
};
