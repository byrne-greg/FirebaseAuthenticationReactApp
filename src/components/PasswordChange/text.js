export default {
  changeYourPassword: "Change your password",
  resetMyPassword: "Reset My Password",
  newPassword: "New Password",
  confirmNewPassword: "Confirm New Password"
};
