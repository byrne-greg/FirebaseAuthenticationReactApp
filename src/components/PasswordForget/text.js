export default {
  emailAddress: "Email Address",
  resetMyPassword: "Reset My Password",
  passwordReset: "Password Reset",
  forgotYourPassword: "Forget your password?",
  forgotPassword: "Forgot Password?"
};
