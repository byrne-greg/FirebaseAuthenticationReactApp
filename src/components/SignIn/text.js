export default {
  signIn: "Sign In",
  emailAddress: "Email Address",
  password: "Password"
};
