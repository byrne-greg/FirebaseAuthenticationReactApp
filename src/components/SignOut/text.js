export default {
  successfulSignOut: "You have successfully signed out",
  signOut: "Sign Out"
};
