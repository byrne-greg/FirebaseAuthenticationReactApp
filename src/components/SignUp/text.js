export default {
  signUp: "Sign Up",
  username: "User Name",
  emailAddress: "Email Address",
  password: "Password",
  confirmPassword: "Confirm Password"
};
