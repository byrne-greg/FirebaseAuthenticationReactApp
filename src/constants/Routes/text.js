export default {
  LANDING: "Landing",
  SIGN_UP: "Sign Up",
  SIGN_IN: "Sign In",
  SIGN_OUT: "Sign Out",
  HOME: "Home",
  ACCOUNT: "Account",
  ADMIN: "Admin",
  PASSWORD_FORGET: "Forget Password",
  DELETED_USER: "Account Deleted"
};
