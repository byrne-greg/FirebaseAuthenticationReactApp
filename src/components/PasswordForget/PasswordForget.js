import React, { useState, useContext } from "react";
import { Link } from "react-router-dom";
import { FirebaseContext } from "../Firebase";
import Routes from "../../constants/Routes";
import text from "./text";

const PasswordForgetPage = () => (
  <div>
    <h1>{text.passwordReset}</h1>
    <PasswordForgetForm />
  </div>
);

const EmailInput = ({ currentValue, onChange }) => (
  <>
    <label id="email-label">{text.emailAddress}</label>
    <input
      aria-labelledby="email-label"
      id="email-input"
      name="email"
      value={currentValue}
      onChange={onChange}
      type="email"
      placeholder={text.emailAddress}
    />
  </>
);

const ErrorAlert = ({ errorText }) => (
  <div>
    <p>{errorText}</p>
  </div>
);

const SubmitButton = ({ isInvalid }) => (
  <button disabled={isInvalid} type="submit">
    {text.resetMyPassword}
  </button>
);

const PasswordForgetForm = () => {
  const [email, setEmail] = useState("");
  const [error, setError] = useState(null);
  const firebase = useContext(FirebaseContext);

  const isInvalid = !email;

  const onSubmit = async event => {
    try {
      await firebase.resetPassword(email);
      setEmail("");
      setError(null);
    } catch (error) {
      setError({ ...error });
    }

    event.preventDefault();
  };

  const handleChange = event => {
    setEmail(event.target.value);
  };

  return (
    <>
      <h2>{text.forgotYourPassword}</h2>
      <form onSubmit={onSubmit}>
        <EmailInput currentValue={email} onChange={handleChange} />
        <SubmitButton isInvalid={isInvalid} />
        {error && <ErrorAlert errorText={error.message} />}
      </form>
    </>
  );
};

const PasswordForgetLink = () => (
  <p>
    <Link to={Routes.PASSWORD_FORGET.url}>{text.forgotPassword}</Link>
  </p>
);

export default PasswordForgetPage;
export { PasswordForgetForm, PasswordForgetLink };
