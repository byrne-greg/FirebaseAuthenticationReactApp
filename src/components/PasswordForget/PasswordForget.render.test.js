import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import { withTestRouter, withMockFirebase } from "../../TestUtil";
import PasswordForget, { PasswordForgetLink } from ".";
import text from "./text";

afterEach(cleanup);

describe("PasswordForget render tests", () => {
  it("page renders as expected", () => {
    // GIVEN

    // WHEN
    const page = render(withTestRouter(<PasswordForget />));

    // THEN
    expect(page.baseElement).toMatchSnapshot();
  });

  it("link renders as expected", () => {
    // GIVEN

    // WHEN
    const link = render(withTestRouter(<PasswordForgetLink />));

    // THEN
    expect(link.baseElement).toMatchSnapshot();
  });

  xit("form renders with error", () => {});
});
