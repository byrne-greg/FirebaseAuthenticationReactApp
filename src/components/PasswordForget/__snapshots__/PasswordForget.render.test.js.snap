// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordForget render tests link renders as expected 1`] = `
<body>
  <div>
    <p>
      <a
        href="/pw-forget"
      >
        Forgot Password?
      </a>
    </p>
  </div>
</body>
`;

exports[`PasswordForget render tests page renders as expected 1`] = `
<body>
  <div>
    <div>
      <h1>
        Password Reset
      </h1>
      <h2>
        Forget your password?
      </h2>
      <form>
        <label
          id="email-label"
        >
          Email Address
        </label>
        <input
          aria-labelledby="email-label"
          id="email-input"
          name="email"
          placeholder="Email Address"
          type="email"
          value=""
        />
        <button
          disabled=""
          id="submit-button"
          type="submit"
        >
          Reset My Password
        </button>
      </form>
    </div>
  </div>
</body>
`;
